src/GafferImage/Unpremultiply.cpp

```diff
@@ -85,11 +97,32 @@ void Unpremultiply::affects( const Gaffer::Plug *input, AffectedPlugsContainer &
 
 void Unpremultiply::hashChannelData( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
 {
-	std::string alphaChannel = alphaChannelPlug()->getValue();
+	std::string alphaChannel;
+	ConstStringVectorDataPtr inChannelNamesPtr;
+	bool ignoreMissingAlpha;
 
-	ChannelDataProcessor::hashChannelData( output, context, h );
+	{
+		ImagePlug::GlobalScope c( context );
+		alphaChannel = alphaChannelPlug()->getValue();
+		inChannelNamesPtr = inPlug()->channelNamesPlug()->getValue();
+		ignoreMissingAlpha = ignoreMissingAlphaPlug()->getValue();
+	}
```

Comment:
In Premultiply, we have a hash pass-through for the `channelName == alphaChannel` case. Might be worth doing the same here while we're at it?

Reply:
Addressed.